backend/src/schema/resolvers/users.js

```diff
@@ -22,6 +22,7 @@ export const getBlockedUsers = async context => {
 }
 
 export const getBlockedByUsers = async context => {
+  if (context.user.role === 'moderator' || context.user.role === 'admin') return []
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-01T22:39:26Z" title="Wednesday, October 2nd 2019, 12:39:26 am +02:00">Oct 2, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -30,6 +30,7 @@ export const getBlockedByUsers = async context => {
     .relationship(userModel.relationships().get('blocked'))
     .to('blocked', userModel)
     .where('blocked.id', context.user.id)
+    .where('blocked.role', 'user')
```


Ah, **that's** the line! Now I get it.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-01T22:46:58Z" title="Wednesday, October 2nd 2019, 12:46:58 am +02:00">Oct 2, 2019</time>_

---


@mattwr18 I would prefer you could add a guard clause and return empty array if `context.user.role > 'user'` or something like that. It took me a while to understand that the additional `where('blocked.role')` is excluding *your own user node* from the result set, thus returning empty array. We could make this more explicit by returning empty array for this case.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-01T22:47:42Z" title="Wednesday, October 2nd 2019, 12:47:42 am +02:00">Oct 2, 2019</time>_

---


A guard clause would also save us one unnecessary database call.